src/__tests__/index-test.js

```diff
@@ -175,10 +179,14 @@ describe("toArray", () => {
     const content = toArray("https://google.com");
     expect(content).toMatchSnapshot();
   });
-  test("url and emoji (no space)", () => {
+  test("url and ascii emoji alias (no space)", () => {
     const content = toArray("https://google.com:)");
     expect(content).toMatchSnapshot();
   });
+  test("url and emoji (no space)", () => {
+    const content = toArray("https://google.com:heart:");
+    expect(content).toMatchSnapshot();
+  });
```

Comment:
These 2 tests should be consistent. Meaning that is think `:)` is part of the URL and thus not matching, `:heart:` should not be matching neither (which is not the case). I think indeed we should see this case as part of the URL as create no match.

Reply:
Let's see what github does:
* `https://example.com:)` https://example.com:)
* `https://example.com/foo/:heart:` https://example.com/foo/:heart:
* `https://example.com/foo/:heart:/bar` https://example.com/foo/:heart:/bar
* `https://example.com/foo:heart:bar` https://example.com/foo:heart:bar

Not exactly ideal either :(

I think the most important situation, which this PR fixes, is the case of short ascii aliases like `:p`, which you really don't want to be replaced in random places in URLs. I think the "full names" like`:heart:` are less likely to cause problems, so I'd be OK with either approach for those.

That would of course need to be fixed in another part of the code:

```javascript
const urls = "https?\\S*";
function getAliasesRegex() {
  return new RegExp(`(?<!${urls}):([${allowedAliasCharacters}]+):`, "g");
}
```

I can add that change to the PR if you want.

Reply:
Right, I think there will always be people on one side of the other. I think for the sake of simplicity in our case and be consistent across all possible edge cases, we should not replace any ascii or full name alias that is part of a URL. Meaning, all the cases above should not convert. If you want to place an emoji next to a URL, you can do it before (with or without space), or after (with space).

How does that sound to you ? cc @tommoor as well

Reply:
Fine with me. I've added a commit for that.